bin/tests/common.rs

```diff
@@ -119,8 +140,15 @@ impl<'a> AgentSettings<'a> {
 pub fn spawn_agent(settings: AgentSettings) -> Child {
     let mut cmd = Command::cargo_bin("logdna-agent").unwrap();
 
-    let ingestion_key =
-        std::env::var("LOGDNA_INGESTION_KEY").expect("LOGDNA_INGESTION_KEY env var not set");
+    let ingestion_key: String;
```

Comment:
Extremely minor, but pretty much everything in rust is an expression, so you can do

```
let ingestion_key = if let Some(key) = settings.ingester_key {
        ingestion_key = key.to_string();
    } else {
        ingestion_key = std::env::var("LOGDNA_INGESTION_KEY")
            .expect("LOGDNA_INGESTION_KEY env var could not be obtained");
    };
```